libs/failsafe-controller/src/main/java/net/jodah/failsafe/SyncFailsafeConfigDelegater.java

```diff
@@ -61,13 +67,134 @@ public SyncFailsafeConfigDelegater(FailsafeConfig<R, ?> master) {
+    return super.onFailedAttempt(
+        new MonitoredContextualResultListener<>(Listeners.of(listener), executor));
+  }
+
+  @Override
+  public SyncFailsafe<R> onFailedAttemptAsync(
+      ContextualResultListener<? extends R, ? extends Throwable> listener,
+      ExecutorService executor) {
+    return super.onFailedAttempt(new MonitoredContextualResultListener<>(listener, executor));
+  }
+
   @Override
-  ListenerRegistry<R> registry() {
-    final ListenerRegistry<R> listenerRegistry = master.registry();
+  public SyncFailsafe<R> onFailureAsync(
+      CheckedBiConsumer<? extends R, ? extends Throwable> listener, ExecutorService executor) {
+    return super.onFailure(
+        new MonitoredContextualResultListener<>(Listeners.of(listener), executor));
+  }
+
+  @Override
+  public SyncFailsafe<R> onFailureAsync(
+      CheckedConsumer<? extends Throwable> listener, ExecutorService executor) {
+    return super.onFailure(
+        new MonitoredContextualResultListener<>(Listeners.of(listener), executor));
+  }
+
+  @Override
+  public SyncFailsafe<R> onFailureAsync(
+      ContextualResultListener<? extends R, ? extends Throwable> listener,
+      ExecutorService executor) {
+    return super.onFailure(new MonitoredContextualResultListener<>(listener, executor));
+  }
+
+  @Override
+  public SyncFailsafe<R> onRetriesExceededAsync(
+      CheckedBiConsumer<? extends R, ? extends Throwable> listener, ExecutorService executor) {
+    registry()
+        .retriesExceeded()
+        .add(new MonitoredContextualResultListener<>(Listeners.of(listener), executor));
+    return this;
+  }
+
+  @Override
+  public SyncFailsafe<R> onRetriesExceededAsync(
+      CheckedConsumer<? extends Throwable> listener, ExecutorService executor) {
+    registry()
+        .retriesExceeded()
+        .add(new MonitoredContextualResultListener<>(Listeners.of(listener), executor));
+    return this;
+  }
 
-    // make sure we use the same registry
-    super.listenerRegistry = listenerRegistry;
-    return listenerRegistry;
+  @Override
+  public SyncFailsafe<R> onRetryAsync(
+      CheckedBiConsumer<? extends R, ? extends Throwable> listener, ExecutorService executor) {
+    return super.onRetry(new MonitoredContextualResultListener<>(Listeners.of(listener), executor));
+  }
+
+  @Override
+  public SyncFailsafe<R> onRetryAsync(
+      CheckedConsumer<? extends Throwable> listener, ExecutorService executor) {
+    return super.onRetry(new MonitoredContextualResultListener<>(Listeners.of(listener), executor));
+  }
+
+  @Override
+  public SyncFailsafe<R> onRetryAsync(
+      ContextualResultListener<? extends R, ? extends Throwable> listener,
+      ExecutorService executor) {
+    return super.onRetry(new MonitoredContextualResultListener<>(listener, executor));
+  }
+
+  @Override
+  public SyncFailsafe<R> onSuccessAsync(
+      CheckedBiConsumer<? extends R, ExecutionContext> listener, ExecutorService executor) {
+    registry()
```

Comment:
ℹ️ There is no onSuccess() method that accepts a generic ContextualResultListener as in the other cases